core/camera/src/main/java/com/google/jetpackcamera/core/camera/CameraSession.kt

```diff
@@ -327,6 +320,28 @@ internal suspend fun processTransientSettingEvents(
     }
 }
 
+context(CameraSessionContext)
+internal fun setZoomScale(camera: Camera, zoomScale: Float) {
+    camera.cameraInfo.zoomState.value?.let { zoomState ->
+        transientSettings.value?.let { transientSettings ->
+            if (!zoomScaleMap.containsKey(transientSettings.primaryLensFacing)) {
+                zoomScaleMap[transientSettings.primaryLensFacing] = 1f
+            }
+            val oldScale = zoomScaleMap[transientSettings.primaryLensFacing]!!
+            val finalScale =
+                (oldScale * zoomScale).coerceIn(
+                    zoomState.minZoomRatio,
+                    zoomState.maxZoomRatio
+                )
+            camera.cameraControl.setZoomRatio(finalScale)
+            zoomScaleMap[transientSettings.primaryLensFacing] = finalScale
```

Comment:
We don't necessarily want to persist the zoom when the lens is flipped. We should reset this to 1f when the camera lens flips, which could happen in CameraXCameraUseCase.